test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs

```diff
@@ -32,15 +32,21 @@ protected HttpSuccessClient()
         {
         }
 
+        /// <summary> Initializes a new instance of HttpSuccessClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
+        public HttpSuccessClient(AzureKeyCredential credential) : this(credential, new Uri("http://localhost:3000"), new HttpSuccessClientOptions())
+        {
+        }
         /// <summary> Initializes a new instance of HttpSuccessClient. </summary>
         /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
         /// <param name="endpoint"> server parameter. </param>
         /// <param name="options"> The options for configuring the client. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> is null. </exception>
-        public HttpSuccessClient(AzureKeyCredential credential, Uri endpoint = null, HttpSuccessClientOptions options = null)
+        /// <exception cref="ArgumentNullException"> <paramref name="credential"/> or <paramref name="endpoint"/> is null. </exception>
+        public HttpSuccessClient(AzureKeyCredential credential, Uri endpoint, HttpSuccessClientOptions options = null)
```

Comment:
Curious: why provide a default value for `options` here?

Reply:
As we agreed in https://github.com/azure/autorest.csharp/issues/2055#issuecomment-1076868591

> If ClientOptions is not the only additional parameter for the overload, we should not make it required.

In this case, we have two optional parameters: `endpoint` and `options`. So I left `options` as optional.

Reply:
I see.  https://github.com/Azure/autorest.csharp/issues/2055#issuecomment-1076868591 was specifically regarding the case where we have global parameters in the swagger, and we [have not finalized the guidance](https://github.com/Azure/autorest.csharp/issues/2106) for that case.

But I understand now - this is because it is a test case and has an optional endpoint.